x-pack/legacy/plugins/monitoring/public/lib/setup_mode.js

```diff
@@ -70,11 +73,41 @@ const notifySetupModeDataChange = (oldData) => {
 
 export const updateSetupModeData = async (uuid, fetchWithoutClusterUuid = false) => {
   const oldData = setupModeState.data;
-  setupModeState.data = await fetchCollectionData(uuid, fetchWithoutClusterUuid);
-  if (get(setupModeState.data, '_meta.isOnCloud', false)) {
+  const data = await fetchCollectionData(uuid, fetchWithoutClusterUuid);
+  setupModeState.data = data;
+  if (get(data, '_meta.isOnCloud', false)) {
     return toggleSetupMode(false); // eslint-disable-line no-use-before-define
   }
   notifySetupModeDataChange(oldData);
+
+  const globalState = angularState.injector.get('globalState');
+  const clusterUuid = globalState.cluster_uuid;
+  if (!clusterUuid) {
+    const liveClusterUuid = get(data, '_meta.liveClusterUuid');
+    const migratedEsNodes = Object.values(get(data, 'elasticsearch.byUuid', {}))
+      .filter(node => node.isPartiallyMigrated || node.isFullyMigrated);
+    if (migratedEsNodes.length > 0) {
+      setNewlyDiscoveredClusterUuid(liveClusterUuid);
```

Comment:
Since you're using `get` to get the `liveClusterUuid` I assume it is not guaranteed to be there? Which could potentially set `globalState.cluster_uuid` to undefined.

I could be totally wrong here though

Reply:
Yea nice catch. I'll make the if: `if (liveClusterUuid && migratedEsNodes.length > 0) {`